src/renderer/vt/vtrenderer.hpp

```diff
@@ -154,9 +154,19 @@ namespace Microsoft::Console::Render
         std::optional<TextColor> _newBottomLineBG{ std::nullopt };
 
         [[nodiscard]] HRESULT _Write(std::string_view const str) noexcept;
-        [[nodiscard]] HRESULT _WriteFormattedString(const std::string* const pFormat, ...) noexcept;
         [[nodiscard]] HRESULT _Flush() noexcept;
 
+        template<typename S, typename... Args>
+        [[nodiscard]] HRESULT _WriteFormatted(S&& format, Args&&... args)
+        try
+        {
+            //auto s = fmt::format(std::forward<S>(format), std::forward<Args>(args)...);
+            //return _Write(s);

```

Comment:
How important is the `_trace.TraceString(str)` and `#ifdef UNIT_TESTING` thing in `_Write`? Just asking...for a friend.

Directly `fmt::format_to` the buffer is *very* fast, because it doesn't create a new `std::string`. But for `_Write` we have to create the `std::string` for the two things I mentioned above.

Reply:
Oh I didn't think of the ETW tracing...

We could do:
```cpp
template<typename S, typename... Args>
[[nodiscard]] HRESULT _WriteFormatted(S&& format, Args&&... args)
try
{
    fmt::basic_memory_buffer<char, 64> buf;
    fmt::format_to(std::back_inserter(buf), std::forward<S>(format), std::forward<Args>(args)...);
    return _Write(buf);
}
```

Reply:
I mean 64 bytes are enough for all our purposes right? We could also reduce it to 32 if we wanted to. The largest string we format is 20 bytes I believe. It doesn't hurt modern CPUs too much though to allocate more memory on the stack. Unless you touch the bytes they're not loaded into the CPU cache after all...

Reply:
Yeah agreed. Stack allocation is much cheaper. The default allocation size of `fmt:format` is 500. I think 64 is reasonable.